src/libsyntax/ast.rs

```diff
@@ -161,10 +227,14 @@ impl GenericArgs {
     }
 }
 
+/// Concrete argument in the sequence of generic args.
 #[derive(Clone, RustcEncodable, RustcDecodable, Debug)]
 pub enum GenericArg {
+    /// `'a` in `Foo<'a>`
     Lifetime(Lifetime),
+    /// `Bar` in `Foo<Bar>`
     Type(P<Ty>),
+    /// `()` in `Foo<()>`
```

Comment:
`()` here is interpreted as a type argument, `Foo<1>` would be a better example.